static/app/views/dashboards/datasetConfig/errors.tsx

```diff
@@ -0,0 +1,240 @@
+  // getTimeseriesSortOptions,
+  getTableSortOptions,
+  getGroupByFieldOptions: getEventsTableFieldOptions,
+  handleOrderByReset,
+  supportedDisplayTypes: [DisplayType.TABLE],
+  getTableRequest: (
+    api: Client,
+    _widget: Widget,
+    query: WidgetQuery,
+    organization: Organization,
+    pageFilters: PageFilters,
+    _onDemandControlContext?: OnDemandControlContext,
+    limit?: number,
+    cursor?: string,
+    referrer?: string,
+    _mepSetting?: MEPState | null
+  ) => {
+    return getEventsRequest(
+      api,
+      query,
+      organization,
+      pageFilters,
+      limit,
+      cursor,
+      referrer
+    );
+  },
+  transformTable: transformEventsResponseToTable,
+  filterAggregateParams,
+};
+
+function getEventsTableFieldOptions(
+  organization: Organization,
+  tags?: TagCollection,
+  _customMeasurements?: CustomMeasurementCollection
+) {
+  return generateFieldOptions({
+    organization,
+    tagKeys: Object.values(tags ?? {}).map(({key}) => key),
+  });
+}
+
+function transformEventsResponseToTable(
+  data: TableData | EventsTableData,
+  _widgetQuery: WidgetQuery
+): TableData {
+  let tableData = data;
+  // events api uses a different response format so we need to construct tableData differently
+  const {fields, ...otherMeta} = (data as EventsTableData).meta ?? {};
+  tableData = {
+    ...data,
+    meta: {...fields, ...otherMeta},
+  } as TableData;
+  return tableData as TableData;
+}
+
+function renderEventIdAsLinkable(data, {eventView, organization}: RenderFunctionBaggage) {
+  const id: string | unknown = data?.id;
+  if (!eventView || typeof id !== 'string') {
+    return null;
+  }
+
+  const eventSlug = generateEventSlug(data);
+
+  const target = eventDetailsRouteWithEventView({
+    orgSlug: organization.slug,
+    eventSlug,
+    eventView,
+  });
+
+  return (
+    <Tooltip title={t('View Event')}>
+      <Link data-test-id="view-event" to={target}>
+        <Container>{getShortEventId(id)}</Container>
+      </Link>
+    </Tooltip>
+  );
+}
+
+function renderTraceAsLinkable(
+  data,
+  {eventView, organization, location}: RenderFunctionBaggage
+) {
+  const id: string | unknown = data?.trace;
+  if (!eventView || typeof id !== 'string') {
+    return null;
+  }
+  const dateSelection = eventView.normalizeDateSelection(location);
+  const target = getTraceDetailsUrl({
+    organization,
+    traceSlug: String(data.trace),
+    dateSelection,
+    timestamp: getTimeStampFromTableDateField(data.timestamp),
+    location,
+  });
+
+  return (
+    <Tooltip title={t('View Trace')}>
+      <Link data-test-id="view-trace" to={target}>
+        <Container>{getShortEventId(id)}</Container>
+      </Link>
+    </Tooltip>
+  );
+}
+
+export function getCustomEventsFieldRenderer(field: string, meta: MetaType) {
+  if (field === 'id') {
+    return renderEventIdAsLinkable;
+  }
+
+  if (field === 'trace') {
+    return renderTraceAsLinkable;
+  }
+
+  return getFieldRenderer(field, meta, false);
+}
```

Comment:
For the function that overlap with errorsAndTransactions, can we just export those and reuse instead of re-implementing here?